### DISCUSSION TITLE
How to get form parameters with special characters

### DISCUSSION THREAD

**Discussion by mingqing6364** [fastapi/fastapi#6215] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import uvicorn
from fastapi import FastAPI, Form, Response

env = {
    "app": "ntunnel_mysql:app",
    "host": "0.0.0.0",
    "port": 8611,
    "workers": 1,
    "debug": True
}

app = FastAPI(debug=env["debug"])


@app.post(path='/mysql')
async def mysql(
        host: str = Form(None),
        port: int = Form(None),
        login: str = Form(None),
        password: str = Form(None),
        db: str = Form(None),
        actn: str = Form(None),
        q: list = Form(None),
        encodeBase64: str = Form(None)
):
    print(host, port, login, password, db, actn, q, encodeBase64)


if __name__ == '__main__':
    uvicorn.run(
        app=env["app"],
        host=env["host"],
        port=env["port"],
        workers=env["workers"],
        reload=env["debug"],
        debug=env["debug"]
    )
```

### Description

Navicat provides an http proxy tool written in php. I want to rewrite the back-end service with the fastapi, but there is a form parameter q [] that cannot be obtained.

The curl request like this
```
curl --location --request POST 'http://127.0.0.1:8611/mysql' \
--header 'Content-type: application/x-www-form-urlencoded' \
--data-urlencode 'actn=Q' \
--data-urlencode 'host=192.168.108.203' \
--data-urlencode 'port=3308' \
--data-urlencode 'db=' \
--data-urlencode 'login=root' \
--data-urlencode 'password=root' \
--data-urlencode 'dbfile=' \
--data-urlencode 'q[]=SELECT 1 a' \
--data-urlencode 'q[]=SELECT 2 b'
```

Python console output like this
```
192.168.108.203 3308 root root None Q [] None
INFO:     127.0.0.1:60883 - "POST /mysql HTTP/1.1" 200 OK
```

I want the output like this
```
192.168.108.203 3308 root root None Q ['SELECT 1 a', 'SELECT 2 b'] None
INFO:     127.0.0.1:60869 - "POST /mysql HTTP/1.1" 200 OK
```

### Operating System

Windows

### FastAPI Version

0.83.0

### Python Version

Python 3.6.8

**Comment by mingqing6364** [comment]
I see this issues：[Query parameters with special characters / cannot create variable #5733](https://github.com/tiangolo/fastapi/issues/5733)
but, form parameters not in the request.

**Comment by andersonrocha0** [comment, marked as answer]
I tried this code and worked for me:

```python
@app.post(path='/mysql')
async def mysql(
        host: str = Form(None),
        port: int = Form(None),
        login: str = Form(None),
        password: str = Form(None),
        db: str = Form(None),
        actn: str = Form(None),
        q: list = Form(None, alias="q[]"),
        encodeBase64: str = Form(None)
):
    print(host, port, login, password, db, actn, q, encodeBase64)
```

I added an **alias** in the "q" attribute and your curl request worked fine. 

This is the print result: 

```bash
192.168.108.203 3308 root root None Q ['SELECT 1 a', 'SELECT 2 b'] None
```

**Comment by mingqing6364** [comment]
> I tried this code and worked for me:
> 
> ```python
> @app.post(path='/mysql')
> async def mysql(
>         host: str = Form(None),
>         port: int = Form(None),
>         login: str = Form(None),
>         password: str = Form(None),
>         db: str = Form(None),
>         actn: str = Form(None),
>         q: list = Form(None, alias="q[]"),
>         encodeBase64: str = Form(None)
> ):
>     print(host, port, login, password, db, actn, q, encodeBase64)
> ```
> 
> I added an **alias** in the "q" attribute and your curl request worked fine.
> 
> This is the print result:
> 
> ```shell
> 192.168.108.203 3308 root root None Q ['SELECT 1 a', 'SELECT 2 b'] None
> ```

thanks.
